src/components/ethContracts.tsx

```diff
@@ -1,144 +1,87 @@
 import { QuestionCircleOutlined } from "@ant-design/icons";
 import { Alert, Button, Card, Input, Result, Tabs } from "antd";
 import "antd/dist/antd.dark.css";
-import React from "react";
+import React, { useState } from "react";
+import { useFeature } from "../hooks/useFeature";
 import { constants } from "../util/helpers";
 import { ContractCardList } from "./ContractCardList";
+import { ContractTable } from "./ContractTable";
 import { PageContent } from "./index";
 import { SearchCard } from "./SearchCard";
 import "./styles.css";
 
-const defaultState = {
-  contract: null,
-  defs: [],
-  success: false,
-  loading: false,
-  customDefs: [],
-  customDefsStr: "",
-};
 const TAB_KEYS = {
   PACK: "1",
   SINGLE_ADDR: "2",
   CUSTOM: "3",
+  ADDED: "4",
 };
 const manualPlaceholder =
   '[{"inputs":[{"internalType":"address[]","name":"_components","type":"address[]"},{"internalType":"int256[]","name":"_units","type":"int256[]"},{"internalType":"address[]","name":"_modules","type":"address[]"},{"internalType":"contract IController","name":"_controller","type":"address"},{"internalType":"address","name":"_manager","type":"address"},{"internalType":"string","name":"_name","type":"string"},';
 
-class EthContracts extends React.Component<any, any> {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      error: null,
-      contract: null,
-      defs: [],
-      success: false,
-      loading: false,
-      //@ts-expect-error
-      tab: TAB_KEYS.PATH,
-      selectedPackKey: "AAVE",
-    };
-
-    this.addDefs = this.addDefs.bind(this);
-    this.onSmartContractAddress = this.onSmartContractAddress.bind(this);
-    this.renderSuccessAlert = this.renderSuccessAlert.bind(this);
-    this.renderPackCard = this.renderPackCard.bind(this);
-    this.renderCustomCard = this.renderCustomCard.bind(this);
-    this.renderSearchCard = this.renderSearchCard.bind(this);
-  }
+const EthContracts = ({ session, isMobile }) => {
+  const [error, setError] = useState(null);
+  const [defs, setDefs] = useState([]);
+  const [customDefs, setCustomDefs] = useState([]);
+  const [customDefsStr, setCustomDefsStr] = useState("");
+  const [success, setSuccess] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [tab, setTab] = useState(TAB_KEYS.PACK);
+  const { CAN_VIEW_CONTRACTS } = useFeature(session);
 
-  onTabChange(key) {
-    this.setState({ tab: key, success: false, error: null, loading: false });
+  function setDefaultState() {
+    setDefs([]);
+    setSuccess(false);
+    setLoading(false);
+    setCustomDefs([]);
+    setCustomDefsStr("");
   }
 
-  onSmartContractAddress(input) {
```

Comment:
Not seeing this code replicated anywhere else... how are you responding to contract addresses now?

Reply:
Maybe related, I put in a contract address and it found the functions. When I clicked to load them nothing happened and I see:

![image](https://user-images.githubusercontent.com/7378490/157550923-88da924a-0536-47ca-b679-cdaa4aaf1d73.png)

![image](https://user-images.githubusercontent.com/7378490/157550944-d1405dc3-6216-4629-bc23-e901c341845c.png)

Reply:
Same problem loading packs.

Reply:
[967d1d9](https://github.com/GridPlus/lattice-manager/pull/227/commits/967d1d92d5c90eefc8e2ab7978b6545f01559bde)

Reply:
Oh and `onSmartContractAddress` was moved to [`SearchCard.tsx`](https://github.com/GridPlus/lattice-manager/blob/967d1d92d5c90eefc8e2ab7978b6545f01559bde/src/components/SearchCard.tsx#L30) but I forgot to remove it here